smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlModelParser.java

```diff
@@ -797,22 +797,29 @@ private void parseMembers(LoadOperation.DefineShape op, MemberParsing memberPars
 
         while (!eof()) {
             if (peek() == '}') {
+                clearPendingDocs();
                 break;
             }
 
-            parseMember(op, definedMembers, memberParsing);
+            boolean consumedPendingDocs = parseMember(op, definedMembers, memberParsing);
 
-            // Clears out any previously captured documentation
-            // comments that may have been found when parsing the member.
-            clearPendingDocs();
+            // Having loaded a default value means we've run the `br()`
+            // parse that will pre-fetch documentation comments. That's
+            // fine, we should just avoid clearing them so the next
+            // member can have them applied properly.
+            if (!consumedPendingDocs) {
+                // Clears out any previously captured documentation
+                // comments that may have been found when parsing the member.
+                clearPendingDocs();
```

Comment:
Can you just do this in parseMember? I don't think it's called by anything but this method. Then the big explanation can go away too.